application/src/main/java/org/togetherjava/tjbot/commands/help/AskCommand.java

```diff
@@ -128,6 +128,7 @@ private RestAction<Message> handleEvent(InteractionHook eventHook, ThreadChannel
         helper.writeHelpThreadToDatabase(author, threadChannel);
         return sendInitialMessage(guild, threadChannel, author, title, category)
             .flatMap(any -> notifyUser(eventHook, threadChannel))
+            .flatMap(any -> threadChannel.pinMessageById(threadChannel.getLatestMessageId()))
```

Comment:
the latest message isnt necessarily the first, is it?

does it run against the cache or does it send a blocking request?

i suspect getLatestMessageID can also fail and maybe return null?

Reply:
> the latest message isnt necessarily the first, is it?

i mean, it sends the first message then i pin it so maybe yes it is the first

> does it run against the cache or does it send a blocking request?

yes it uses cache i believe

> i suspect getLatestMessageID can also fail and maybe return null?

it is annotated with nonnull, it returns 0 instead of null

Reply:
I see. It should be safe in 99% of cases. Its possible though that it runs into an error message if its `0`. We could:
* add an if-else, log a warning if its 0 and not call `pinMessageById` then

To make that readable, put the stuff into a helper method like `pinFirstMessage`